ui/src/components/Header.tsx

```diff
@@ -30,13 +31,17 @@ type HeaderProps = {
   drawerWidth?: number;
   currentPage?: string | null;
   breadcrumbItem?: React.ReactNode;
+  inRepo?: boolean;
+  setShareOpen?: () => void;
 };
 
 export const Header: React.FC<HeaderProps> = ({
   open = false,
   drawerWidth = 0,
   currentPage = null,
   breadcrumbItem = null,
+  inRepo = false,
+  setShareOpen = () => {},
```

Comment:
1. It seems that `setShareOpen` doesn't need to be passed in.  You could call `useStore` to get it.
2. `inRepo` isn't general enough. I would use a `buttonItem=null` argument and pass in a ReactNode (just like the `breadcrumbItem` above). This way, the repo sharing UI & logic are not leaked into `Header`'s implementation.

E.g.,

```ts
// repo.tsx
function ShareButton() {
  const setShareOpen = useStore(...);
  return <Box><Button onClick={()=>setShareOpen(true)}>...</Button></Box>
}
function RepoWrapper() {
  return <Box> ... <Header buttonItem={<ShareButton/>}/></Box>
}

// Header.tsx
function Header({..., buttonItem=null}) {
  return <Box> ... {buttonItem} ... </Box>
}
```